src/main/kotlin/it/pagopa/wallet/eventdispatcher/service/WalletUsageService.kt

```diff
@@ -0,0 +1,46 @@
+package it.pagopa.wallet.eventdispatcher.service
+
+import it.pagopa.generated.wallets.model.ClientId
+import it.pagopa.wallet.eventdispatcher.api.WalletsApi
+import it.pagopa.wallet.eventdispatcher.domain.WalletUpdateUsageError
+import it.pagopa.wallet.eventdispatcher.utils.Tracing
+import java.time.OffsetDateTime
+import java.util.*
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import reactor.core.publisher.Mono
+
+@Component
+class WalletUsageService(private val walletsApi: WalletsApi) {
+
+    private val log = LoggerFactory.getLogger(WalletUsageService::class.java.name)
+
+    fun updateWalletUsage(
+        walletId: String,
+        clientId: ClientId,
+        usedAt: OffsetDateTime
+    ): Mono<Unit> {
+        log.info("Updating wallet usage walletId: [{}], client: [{}]", walletId, clientId)
+        return walletsApi
+            .updateWalletUsage(UUID.fromString(walletId), clientId, usedAt)
+            .filter { it.statusCode.is2xxSuccessful }
```

Comment:
as per the comment on client this filter is always true since KO error codes will be mapped to Mono.error breaking the chain before filter is evaluated

Reply:
mapping to a proper Exception in an `onErrorMap` method can solve this problem making this filter method useless

Reply:
as per implementation there is no specific error code handling to be perfomed here (just a check if api call is OK/KO) so the above hints should be sufficients.
Just as future hint, if you have to handle every http error codes manually you can try [this approach](https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/blob/2131b5b709cfc5165687604f568a5940da1e8dfa/src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/client/NotificationsServiceClient.kt#L38-L63)